packages/next-international/src/i18n/create-i18n.ts

```diff
@@ -1,21 +1,23 @@
 import { createContext } from 'react';
-import type { Locales, LocaleContext } from '../types';
+import type { Locales as LocalesBase, LocaleContext } from '../types';
 import type { BaseLocale } from 'international-types';
 import { createDefineLocale } from './create-define-locale';
 import { createGetLocaleProps } from './create-get-locale-static-props';
 import { createI18nProvider } from './create-i18n-provider';
 import { createUseChangeLocale } from './create-use-change-locale';
 import { createUsei18n } from './create-use-i18n';
 import { createScopedUsei18n } from './create-use-scoped-i18n';
+import { createUseCurrentLocale } from './create-use-current-locale';
 
-export function createI18n<Locale extends BaseLocale>(locales: Locales) {
+export function createI18n<Locale extends BaseLocale, Locales extends LocalesBase = LocalesBase>(locales: Locales) {
```

Comment:
I don't think we need this generic, as we can already have access to the type of the locales using the `locales` parameter